### ISSUE TITLE
Sign of uncertainties in Dimension3d

### ISSUE THREAD

**Issue by ainar** [OpenSimulationInterface/open-simulation-interface#316]
https://github.com/OpenSimulationInterface/open-simulation-interface/blob/a8fb7ccebe22c14662a86b5649077785ed5adeb3/osi_common.proto#L92

On this line, the documentation let the uncertainties on Dimension3d values be negative. However an uncertainty, according to the [RMSE](https://en.wikipedia.org/wiki/Root-mean-square_deviation) formula, can only be positive. Is it an error or intentional?

**Comment by CarloVanDriestenBMW** [comment]
Interesting finding! Needs to be checked

**Comment by ainar** [comment]
By going further, I notice that actually the problem can be bigger and is generic to every uncertainty.

It is difficult to characterize an error/tolerance on a measurement and this depend on a lot of more factors, how many depending on what you are exactly defining.
For example, if a Dimension3d message characterize the dimension of a car, its tolerance depends on environmental conditions, the temperature of the car, its materials, etc.

Depending on how precise you want OSI to be, these informations should be implemented in a specific new message for each kind of tolerance.

**Comment by jdsika** [comment]
@PhRosenberger the definition of uncertainties is also a major part in the ISO23150 and should be clarified in #367

**Comment by PhRosenberger** [comment]
> 
> 
> @PhRosenberger the definition of uncertainties is also a major part in the ISO23150 and should be clarified in #367

I agree.

**Comment by jdsika** [comment]
Could you have a look at his @FlorianMueller87, please?

**Comment by FlorianMueller87** [comment]
I agree that uncertainties could be positive and negative. The ISO defines them with real value types, see e.g. A.1.23 of the ISO.
The ISO defines the error model in B.4.1 which includes trueness and precision, B.4.1.3.2.
For the harmonisation with ADI an error-model would also be necessary.

**Comment by yash-shah-asam** [comment]
@thomassedlmayer Could you please look at this in the sensor modeling group?

**Comment by PhRosenberger** [comment]
I will close this issue, as it was answered by @FlorianMueller87 years ago.
